resources/2024-02-output-resources.md

```diff
@@ -0,0 +1,210 @@
+# Adding ProvisioningStatus field to the OutputResource
+
+* **Status**: In Progress
+* **Author**: Nithya Subramanian (@nithyatsu)
+
+## Overview
+
+Output resources are the actual platform specific resources that get deployed in order to deploy a Radius resource. For example to deploy a container as part of radius application deployment on kubernetes, we deploy a kubernetes ServiceAccount, Role, RoleBinding and Deployment Output resources.
+
+There are a couple of problems related to how we manage output resources today -
+
+1. Output resources do not have a ProvisioningStatus property. Instead, if any of the output resource deployment fails, we set the Provisioning state of the corresponding radius resource accordingly and do not store the output resources. With this approach, we lose the information on which output resources failed deployment since we aggregate failures into the owning Radius resource's provisioningStatus property. 
+
+2. We do not record any of the output resources as part of the radius resource whose deployment failed, although we deployed them with an errored status. This leads to orphaned resources which should be manually cleaned up.
+
+This design proposes addition of ProvisioningStatus to outputResources so that, 
+once we add changes to track these resources in case of both successful and failed deployment to avoid orphaning resources, there will be clearer indication of what lead to failed deployment to the user. 
+
+
+## Terms and definitions
+
+| Term | Definition |
+|---|---|
+| OutputResource | one or more platform specific resources that make up a Radius resource |
+| Resource | Radius resource |
+| ProvisioningStatus | Indicates whether the resource was provisioned successfully |
+
+
+
+## Objectives
+
+> **Issue Reference:**
+
+<https://github.com/radius-project/radius/issues/7052>
+
+### Goals
+
+The goal is to record the output resources whether or not deployment succeeds, so that radius can still manage the cleanup of resources it failed to deploy successfully. 
```

Comment:
This is independent of the API change right?

Reply:
yes.. I was just trying to capture what lead to this.